src/Components/SuperAdmins/index.js

```diff
@@ -22,90 +24,46 @@ const SuperAdmins = () => {
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/super-admins`);
-      const data = await response.json();
-      saveSuperAdmins(data.data);
-      setLoading(false);
-    } catch (error) {
-      console.error(error);
-    }
-  };
-
   useEffect(() => {
-    fetchData();
+    dispatch(getSuperAdmins());
   }, []);
 
-  const deleteItem = (_id) => {
-    saveSuperAdmins([...superAdmins.filter((listItem) => listItem._id !== _id)]);
-  };
-
-  const addItem = (body) => {
-    const newSuperAdmin = {
-      _id: body._id,
-      firstName: body.firstName,
-      lastName: body.lastName,
-      phone: body.phone,
-      email: body.email,
-      password: body.password,
-      active: body.active
-    };
-    saveSuperAdmins([...superAdmins, newSuperAdmin]);
-  };
-
-  const editItem = (data) => {
-    const superAdminUpdated = superAdmins.map((superAdmin) => {
-      if (superAdmin._id === data._id) {
-        return data;
-      } else {
-        return superAdmin;
-      }
-    });
-    saveSuperAdmins(superAdminUpdated);
-  };
-
   const onClick = () => {
     setShowForm(true);
-    setMethod('POST');
   };
 
-  return loading ? (
-    <Loader show={true} />
-  ) : (
-    <section className={styles.container}>
-      <ListSuperAdmin
-        list={superAdmins}
-        deleteItem={deleteItem}
-        setShowModal={setShowModal}
-        setShowTitle={setShowTitle}
-        setShowForm={setShowForm}
-        setPreviousSuperAdmin={setPreviousSuperAdmin}
-        setLoading={setLoading}
-        setMethod={setMethod}
-      />
-      <Form
-        showForm={showForm}
-        setShowForm={setShowForm}
-        previousSuperAdmin={previousSuperAdmin}
-        setPreviousSuperAdmin={setPreviousSuperAdmin}
-        setShowModal={setShowModal}
-        setShowTitle={setShowTitle}
-        editItem={editItem}
-        addItem={addItem}
-        setLoading={setLoading}
-        method={method}
-      />
-      <Button onClick={onClick}> Add Super Admin</Button>
-      <Modal
-        isOpen={showModal}
-        handleClose={() => {
-          setShowModal(false);
-        }}
-      >
-        <p>{showTitle}</p>
-      </Modal>
-    </section>
+  const handleClose = () => {
+    setShowModal(false);
+  };
+
+  return (
+    <>
+      {isLoading ? (
+        <Loader show={true} />
+      ) : (
+        <section className={styles.container}>
+          <ListSuperAdmin
+            setShowForm={setShowForm}
+            setPreviousSuperAdmin={setPreviousSuperAdmin}
+            setShowModal={setShowModal}
+            setChildrenModal={setChildrenModal}
+          />
+          <Form
+            showForm={showForm}
+            setShowForm={setShowForm}
+            previousSuperAdmin={previousSuperAdmin}
+            setPreviousSuperAdmin={setPreviousSuperAdmin}
+            setShowModal={setShowModal}
+            setChildrenModal={setChildrenModal}
+          />
+          <Button onClick={onClick}> Add Super Admin</Button>
```

Comment:
Aca podrias borrarr la funcion de onClick ya que solamente le estas pasando el valor del setShowForm y ejecutar ese estado como funcion directamente
```suggestion
          <Button onClick={() => setShowForm(true)}> Add Super Admin</Button>
```